<launch>

    <arg name="video_device_right" default="/dev/video0" />
    <arg name="video_device_left" default="/dev/video2" />

    <arg name="image_width" default="640" />
    <arg name="image_height" default="480" />

    <arg name="right_image_namespace" default="right" />
    <arg name="left_image_namespace" default="left" />

    <arg name="right_image_node" default="right" />
    <arg name="left_image_node" default="left" />

    <arg name="image_topic_name_right" default="image_raw" />
    <arg name="camera_info_topic_name_right" default="camera_info" />

    <arg name="image_topic_name_left" default="image_raw" />
    <arg name="camera_info_topic_name_left" default="camera_info" />



    <arg name="left_camera_name" default="left" />
    <arg name="right_camera_name" default="right" />

    <arg name="left_camera_frame_id" default="left_camera" />
    <arg name="right_camera_frame_id" default="right_camera" />

    <arg name="framerate" default="30" />

    <arg name="camera_info" default="false" />
    <arg name="camera_info_path" default="stereo_camera_info" />

    <node ns="$(arg right_image_namespace)" name="$(arg right_image_node)" pkg="libuvc_camera" type="camera_node" output="screen" >
      <!-- Parameters used to find the camera -->
      <param name="vendor" value="0x046d"/>
      <param name="product" value="0x0825"/>
      <param name="serial" value="0A1DCD50"/>
      <!-- If the above parameters aren't unique, choose the first match: -->
      <param name="index" value="0"/>

      <!-- Image size and type -->
      <param name="width" value="$(arg image_width)" />
      <param name="height" value="$(arg image_height)" />
      <!-- choose whichever uncompressed format the camera supports: -->
      <param name="video_mode" value="yuyv"/> <!-- or yuyv/nv12/mjpeg -->
      <remap from="/usb_cam/image_raw" to="$(arg image_topic_name_right)" />
      <param name="frame_rate" value="$(arg framerate)"/> 
      <param name="timestamp_method" value="start"/> <!-- start of frame -->
      <param name="camera_info_url" value="file:///tmp/cam.yaml"/>
    
      <param name="auto_exposure" value="3"/> <!-- use aperture_priority auto exposure -->
      <param name="auto_white_balance" value="false"/>
    </node>
    
    <node ns="$(arg left_image_namespace)" name="$(arg left_image_node)" pkg="libuvc_camera" type="camera_node" output="screen" >
      <!-- Parameters used to find the camera -->
      <param name="vendor" value="0x046d"/>
      <param name="product" value="0x0825"/>
      <param name="serial" value="0A6BCD50"/>
      <!-- If the above parameters aren't unique, choose the first match: -->
      <param name="index" value="0"/>

      <!-- Image size and type -->
      <param name="width" value="$(arg image_width)" />
      <param name="height" value="$(arg image_height)" />
      <!-- choose whichever uncompressed format the camera supports: -->
      <param name="video_mode" value="yuyv"/> <!-- or yuyv/nv12/mjpeg -->
      <remap from="/usb_cam/image_raw" to="$(arg image_topic_name_left)" />
      <param name="frame_rate" value="$(arg framerate)"/> 
      <param name="timestamp_method" value="start"/> <!-- start of frame -->
      <param name="camera_info_url" value="file:///tmp/cam.yaml"/>
    
      <param name="auto_exposure" value="3"/> <!-- use aperture_priority auto exposure -->
      <param name="auto_white_balance" value="false"/>
    </node>
    
</launch>
